packages/zoe/src/zoe.js

```diff
@@ -144,16 +144,24 @@ import { makeTables } from './state';
  * { Asset: amountMath.make(5), Price: amountMath.make(9) }
  *
  * @typedef {AmountKeywordRecord[]} AmountKeywordRecords
+ */
+
+/**
+ * @typedef {Object} Timer
+ * @typedef {number} Deadline
+ *
+ * @typedef {{waived:null}} Waived
+ * @typedef {{onDemand:null}} OnDemand
  *
- * @typedef {Object} ExitRule
+ * @typedef {{afterDeadline:{timer:Timer, deadline:Deadline}}} AfterDeadline
+ *
+ * @typedef {(Waived|OnDemand|AfterDeadline)} ExitRule
```

Comment:
This is an improvement, but in coveredCall, VSCode can't quite figure out ExitRule.

Reply:
I tried it out in a new `t.js`, and this works just fine.  So, maybe something is conflicting elsewhere?